pkg/grpc/api.go

```diff
@@ -1309,12 +1302,46 @@ func (a *api) SubscribeConfigurationAlpha1(request *runtimev1pb.SubscribeConfigu
 		serverStream: configurationServer,
 	}
 
-	// TODO(@laurence) deal with failed subscription and retires
-	_ = store.Subscribe(context.Background(), req, handler.updateEventHandler)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	for _, k := range unsubscribedKeys {
-		a.configurationSubscribe[fmt.Sprintf("%s||%s", request.StoreName, k)] = true
+	// TODO(@laurence) deal with failed subscription and retires
+	err = store.Subscribe(ctx, req, handler.updateEventHandler)
+	if err != nil {
+		err = status.Errorf(codes.InvalidArgument, messages.ErrConfigurationSubscribe, req.Keys, request.StoreName, err.Error())
+		apiServerLogger.Debug(err)
+		return err
 	}
+	stop := make(chan struct{})
+	a.configurationSubscribeLock.Lock()
+	a.configurationSubscribe[getConfigSubscribeUniqueKey(request.GetStoreName(), request.GetKeys())] = stop
```

Comment:
isn't this changing the unique map key pattern? 
If my understanding is correct this is changing the map key to "storename||key1 key2 key3" etc. since request.GetKeys() is a slice. 
If that is the scenario, there will be no option to subscribe and unsubscribe to individual keys right ?

Previously it was "storeName||key1", "storeName||key2" etc ...

Reply:
This is worth considering I think. Considering that we start subscribing by a GRPC stream request, in which multiple keys may be subscribed at the same time, and subscribing and unsubscribing are the entire GRPC Stream dimension, So here will be the key design for "storename | | key1 key2 key3" pattern
@mukundansundar

Reply:
@LaurenceLiZhixin it might be confusing for users to know they need to include all the keys they want to subscribe/unsubscribe to. Since we don't expect big slices (ie. thousands) here, can we change the key to only contain the store name and then keep and have a data structure that encapsulates the keys as the value?

Reply:
If the key is storeName so, I think the unsubscribe API can only unsubscribe specific storeName's subscription. And `UnSubscribeConfigurationRequest` contains only storeName fields. After user calling Unsubscribe API, all blocked grpc-stream of the storeName's subscription are released.

In this way, we need to maintain two maps, one to hold the map of storeName to encapsulated keys data structure and one to hold the stop channel of this storeName that grpc-stream subscribing to.
@yaron2

Reply:
> In this way, we need to maintain two maps, one to hold the map of storeName to encapsulated keys data structure and one to hold the stop channel of this storeName that grpc-stream subscribing to.

Currently, the proto API in this PR allows users to unsubscribe from individual keys, because you have a store name and a list of keys. We need to ensure this is doable. This can be either done with two maps or by keeping the key as `storename||keyX`, where `keyX` is a single key.